src/context/WalletProvider/NativeWallet/NativeSeed/hooks/useRevocableSeed/useRevocableSeed.ts

```diff
@@ -0,0 +1,47 @@
+import { EncryptedWallet } from '@shapeshiftoss/hdwallet-native/dist/crypto'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+
+export const useRevocableSeed = (encryptedWallet?: EncryptedWallet) => {
+  const [generating, setIsGenerating] = useState(true)
+
+  const revocableSeed = useMemo(
+    () =>
+      Proxy.revocable({} as { seed?: string }, {
+        get: (target, name) => (name === 'seed' ? target.seed : undefined),
+        set: (target, name, value) => {
+          if (name === 'seed') {
+            Object.defineProperty(target, name, {
+              enumerable: false,
+              configurable: false,
+              writable: false,
+              value
+            })
+
+            return true
+          }
+          return false
+        }
+      }),
+    []
+  )
+
+  const generate = useCallback(async () => {
+    setIsGenerating(true)
+    if (encryptedWallet && !encryptedWallet.encryptedWallet) {
+      try {
+        await encryptedWallet.createWallet()
+        revocableSeed.proxy.seed = await encryptedWallet.decrypt()
+      } catch (error) {
+        console.error('Error creating wallet', error)
+      }
+      setIsGenerating(false)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [encryptedWallet?.encryptedWallet, revocableSeed])
+
+  useEffect(() => {
+    generate()
+  }, [generate])
+
+  return { revocableSeed, loading: generating }
```

Comment:
I think we have an opportunity here to make it a bit easier.

We could do
```ts
return {
  getSeed: async () => {
    try {
      await generate()
      return revocableSeed.proxy.seed
    } catch (e) {
      // seed has been revoked
      return null // or maybe we generate another seed
    },
    revoke: revockableSeed.revoke
}
```

Reply:
So you are thinking that the code consuming the hook we would need to do this outside of the hook? 
```js
const { getSeed, revoke } = useRevocableSeed()

useEffect(() => {
  void (async () => {
    const s = await getSeed()
    setSeed(s)
  })
}, [])
```
It seems like at that point maybe this should not be a hook and should just be a helper function.

Or we could allow the hook to handle the generation on mount and then we just update it to return something a little more simple
```js
return { seed: revocableSeed.proxy.seed, revoke: revocableSeed.revoke }
```

Reply:
The whole point is to NOT put the seed into state variables, so I wouldn't do that. Since it's just a simple function, you can just do `getSeed` where you need it. If making it async is a problem, then you could still do the `useEffect(() => generate()` and return a function that returns the value or catches the error and returns null.

Once a Proxy is revoked it'll always throw an error.

Reply:
Ah yea that makes a lot of sense. Thank you and updated with new changes.